hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoDeleteFeeCalculator.java

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: Apache-2.0
+package com.hedera.node.app.service.token.impl.handlers;
+
+import static java.util.Objects.requireNonNull;
+import static org.hiero.hapi.fees.FeeScheduleUtils.lookupServiceFee;
+
+import com.hedera.hapi.node.base.HederaFunctionality;
+import com.hedera.hapi.node.base.SubType;
+import com.hedera.hapi.node.transaction.Query;
+import com.hedera.hapi.node.transaction.TransactionBody;
+import com.hedera.node.app.spi.fees.AbstractSimpleFeeCalculator;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import org.hiero.hapi.fees.FeeResult;
+
+/** Calculates CryptoDelete fees. Per HIP-1261, only uses SIGNATURES extra. */
+public class CryptoDeleteFeeCalculator extends AbstractSimpleFeeCalculator {
+
+    @Override
+    @NonNull
+    public FeeResult calculateTxFee(@NonNull final TransactionBody body, @Nullable final TxContext context) {
+        requireNonNull(body, "body");
+
+        // Build usage mapper - fluent API makes fee structure explicit
+        final var usageMapper =
+                usageBuilder().withSignatures(context.numTxnSignatures()).build();
+
+        // Get fee schedule
+        final var feeCalculator = context.feeCalculatorFactory().feeCalculator(SubType.DEFAULT);
+        final var feeSchedule = feeCalculator.getSimpleFeesSchedule();
+        final var serviceDef = lookupServiceFee(feeSchedule, HederaFunctionality.CRYPTO_DELETE);
+
+        // Calculate fees
+        long nodeFee = feeSchedule.node().baseFee()
+                + calculateExtraFees(feeSchedule.node().extras(), feeSchedule, usageMapper);
+
+        long serviceFee = serviceDef.baseFee() + calculateExtraFees(serviceDef.extras(), feeSchedule, usageMapper);
+
+        long networkFee = calculateNetworkFee(nodeFee, feeSchedule.network().multiplier());
+
+        // Return result
+        final var result = new FeeResult();
+        result.node = nodeFee;
+        result.network = networkFee;
+        result.service = serviceFee;
+        return result;
+    }
+
+    @Override
+    @NonNull
+    public FeeResult calculateQueryFee(@NonNull final Query query, @Nullable final QueryContext context) {
```

Comment:
same here we dont need this. This will only be overridden in queries and we call this method only in `QueryWorkFlowImpl`